### DISCUSSION TITLE
error message with using pr-pull label in homebrew tap

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#4938] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
na
```

### Output of `brew doctor`

```shell
na
```

### Description of issue

built some bottles for the homebrew-freecad tap late last evening. assigned the `pr-pull` label to the pr i submitted to build the bottles, and am now getting the below error message. something i believe have not seen before using the pr-pull label 😕

```
==> aria2
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
Error: Failure while executing; `/usr/local/bin/brew install --formula skopeo` exited with 1.
Warning: Removed Sorbet lines from backtrace!
```

link to the entire job run below,

https://github.com/orgs/Homebrew/discussions/4935#discussioncomment-7711166

link to the original pr

https://github.com/FreeCAD/homebrew-freecad/actions/runs/7049670165/job/19188671058?pr=446

**Comment by ZhongRuoyu** [comment, marked as answer]
The error seems to be from https://github.com/FreeCAD/homebrew-freecad/actions/runs/7049670165/job/19188671058?pr=446#step:4:462:

```
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/2to3
Target /usr/local/bin/2to3
already exists. You may want to remove it:
  rm '/usr/local/bin/2to3'
```

This is an issue with the GitHub Actions' macOS runner image; some of the Python symlinks are overwritten. Running `brew unlink python && brew link --overwrite python` before `pr-pull` is a workaround ([it's also used in `Homebrew/brew`'s CI](https://github.com/Homebrew/brew/commit/298003b18191eb55f58f19ca1b5d0af45e37632a)).

Is there any specific reason you are running the pr-pull action on macOS? This problem shouldn't happen on `ubuntu-latest`.

**Reply by ipatch** [reply]
weird it's been using macos-latest for over two years from what i'm seeing (looking at the commit history for the workflow file). thanks for the suggestion/fix 🙏. i guess i'll have to update workflow file to use `ubuntu-latest`.

the downside to that though once i make that change and rebase my PR i believe the way homebrew / test-bot is setup that it will trigger a rebuild of my bottles even though nothing has changed in the actual PR other than rebasing a commit that changes a different file ie the workflow file. which on my self-hosted runners it takes ~ 1hr to build a bottle 😭
